prometheus/exporter/network.py

```diff
@@ -0,0 +1,163 @@
+        desSplit = desStr.split("B")
+        srcAddress = re.findall(r'[0-9]+(?:\.[0-9]+){3}:[0-9]+', srcSplit[0])
+        srcIPPort = ""
+
+        if len(srcAddress) > 0:
+            srcIPPort = srcAddress[0]
+
+        desAddress = re.findall(r'[0-9]+(?:\.[0-9]+){3}:[0-9]+', desSplit[0])
+        desIPPort = ""
+
+        if len(desAddress) > 0:
+            desIPPort = desAddress[0]
+
+        connection = srcIPPort + "|" + desIPPort
+        lastOutDataSize = convert_to_byte(srcSplit[-3].strip())
+        lastInDataSize = convert_to_byte(desSplit[-3].strip())
+        connectionDic[connection] = {"inSize": lastInDataSize, "outSize": lastOutDataSize, "src": srcIPPort, "out": desIPPort}
+        index += 2
+    return connectionDic
+
+def iftop():
+    try:
+        iftopCMD = "iftop -t -P -s 1 -L 10000 -B -n -N"
+        iftopResult = subprocess.check_output([iftopCMD], shell=True)
+        result = parse_iftop(iftopResult)
+        return result
+    except: 
+        exception = sys.exc_info()
+        for e in exception:
+            logger.error("exporter iftop error {}".format(e))
+        connectionDic = {}
+        return connectionDic
+
+def parse_lsof(stats):
+    connections = {}
+    data = stats.splitlines()
+    data.pop(0)
+    index = 0
+
+    for ditem in data:
+        srcDes = re.findall(r'[0-9]+(?:\.[0-9]+){3}:[0-9]+', ditem)
+        srcStr = ""
+        desStr = ""
+        srcDesStr = ""
+
+        if 2 == len(srcDes):
+            srcStr = srcDes[0]
+            desStr = srcDes[1]
+            srcDesStr = srcStr + "|" + desStr
+            connections[srcDesStr] = {"src": srcStr, "des": desStr}
+    return connections
+
+def lsof(containerPID):
+    try:
+        lsofCMD = "infilter {} /usr/bin/lsof -i -n".format(containerPID)
+        isofResult = subprocess.check_output([lsofCMD], shell=True)
+        result = parse_lsof(isofResult)
+        return result
+    except:
+        exception = sys.exc_info()
+        for e in exception:
+            logger.error("exporter lsof error {}".format(e))
+        connections = {}
+        return connections
+
+def acc_per_container_network_metrics(connectionDic, pid):
+    try:
+        connections = lsof(pid)
+        accInBytes = 0
+        accOutBytes = 0
+
+        for conn in connections:
+            inBytes = 0
+            outBytes = 0
+            if conn in connectionDic:
+                inBytes = connectionDic[conn]["inSize"]
+                accInBytes += inBytes
+            if conn in connectionDic:
+                outBytes = connectionDic[conn]["outSize"]
+                accOutBytes += outBytes
+        return accInBytes, accOutBytes
+    except: 
+        exception = sys.exc_info()
+        for e in exception:
+            logger.error("exporter acc_per_container_network_metrics error {}".format(e))
+        accInBytes = 0
+        accOutBytes = 0
+        return 0, 0
+
+def main(argv):
+    timeSleep = 3
+    iter = 0
+    pid = [argv]
+    while(True):
```

Comment:
well, do not around True with parentheses

Reply:
will update it